src/routes/bkpr/components/RoutingFeesHistory.svelte

```diff
@@ -0,0 +1,293 @@
+          const value = channelID ? routingDate[channelID] : routingDate.total
+          day.y = parseInt(
+            convertValue({
+              value,
+              from: BitcoinDenomination.msats,
+              to: $settings$.bitcoinDenomination
+            }) as string
+          )
+        }
+      }
+    }
+
+    return dayData.map((day) => day.y)
+  }
+
+  // Generate chart data using chosen range of dates & channels
+  function updateChartDatasets() {
+    chart.data.datasets = []
+    const dayData = Array.from(chartDatesSliced, (date) => ({ x: date, y: 0 }))
+    // A line for each selected channel on the chart
+    if (activeChannelIDs.length) {
+      activeChannelIDs.forEach((channelID, i) => {
+        const data = getChartData(dayData, channelID)
+
+        chart.data.datasets.push({
+          label: truncateValue(channelID, 3),
+          data,
+          fill: false,
+          borderColor: colors[i % colors.length]
+        })
+      })
+      // A single line for total fees
+    } else {
+      const data = getChartData(dayData)
+
+      chart.data.datasets.push({
+        label: 'Total',
+        data,
+        fill: false
+      })
+    }
+
+    chart.update()
+  }
+
+  async function renderChart() {
+    const { Chart } = await import('chart.js/auto')
+
+    const ticksColor = $settings$.darkmode ? 'white' : 'black'
+
+    if (chartEl) {
+      chart = new Chart(chartEl, {
+        type: 'line',
+        data: {
+          labels: [],
+          datasets: []
+        },
+        options: {
+          ...(device.type === 'mobile' && { aspectRatio: 1.1 }), // make chart more readible on mobile
+          scales: {
+            x: {
+              display: true,
+              ticks: {
+                color: ticksColor
+              },
+              title: {
+                color: ticksColor,
+                display: true,
+                text: 'DATE'
+              }
+            },
+            y: {
+              display: true,
+              beginAtZero: true,
+              ticks: {
+                color: ticksColor
+              },
+              title: {
+                color: ticksColor,
+                display: true,
+                text: $settings$.bitcoinDenomination.toLocaleUpperCase()
+              }
+            }
+          }
+        }
+      })
+    }
+  }
+
+  $: routingEvents = ($incomeEvents$.data || []).filter(
+    (event) => event.tag === 'routed' && Number(event.credit_msat) > 0
+  )
+  $: noRoutingFees = !routingEvents.length
+  $: routingDates = formatRoutesByDate(routingEvents)
+  $: channelIDs = new Set(routingEvents?.map((event) => event.account))
+
+  // Generate chartDates & initalize chart range
+  $: if (routingEvents.length) {
+    chartDates = xAxisDates(
+      new Date(routingEvents[0]?.timestamp * 1000), // First event date
+      new Date(routingEvents[routingEvents.length - 1]?.timestamp * 1000) // Last event date
+    )
```

Comment:
```suggestion
    chartDates = xAxisDates(
      new Date(routingDates[0].date), // First event date
      new Date(routingDates[routingDates.length - 1].date) // Last event date
    )
```

We want to use the calculated dates for start and end. Using the routing events was missing routes from today's date when testing.